### PR TITLE
Add roadmap file

### PR DESCRIPTION
**Pull request by faxu**
**Description**: Add roadmap file describing current and future investements

**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

File: docs/Roadmap.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+### Continued Performance Optimizations
+Performance is a key focus for ONNX Runtime. From latency to memory utilization to CPU usage, we are constantly seeking strategies to deliver the best performance. Although DNNs are rapidly driving research areas for innovation, we acknowledge that in practice, many companies and developers are still using traditional ML frameworks for reasons ranging from expertise to privacy to legality. As such, ONNX Runtime is focused on improvements and support for both DNNs and traditional ML. 
+
+#### Examples of projects the team is working on:
+* Improvements to batch processing for scikit-learn models
+* More quantization support
+* Multithreading
+* Graph optimizations
+* Intelligent graph partitioning to maximize the value of different accelerators
+* Optimizations for IoT Edge devices
```
Comment (pranavsharma) [inline docs/Roadmap.md]: Is mobile covered under Android EP?

Comment (faxu) [reply docs/Roadmap.md]: it's covered under the platforms section for Android and iOS. Also updated this section to include mobile in addition to IoT Edge and the perf optimization investments may be similar for these smaller-footprint devices

File: docs/Roadmap.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
+# ONNX Runtime Roadmap
+ONNX Runtime is an active, fast-paced project backed by a strong team of deeply technical Microsoft engineers along with a worldwide 
+community of partners and contributors. This roadmap summarizes the pending investments identified by the team to continually grow 
+ONNX Runtime as a robust, versatile, and high performance engine for DNN and traditional ML models.
```
Comment (pranavsharma) [inline docs/Roadmap.md]: high performance *inferencing* engine?

File: docs/Roadmap.md [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+## High Level Goals
+ONNX Runtime fully aligns with the goals of the [ONNX](https://onnx.ai/) community to bring openness and interoperability to the AI industry. 
+ONNX is a standardization specification, and ONNX Runtime is an inference engine for models in the ONNX format to efficiently productionize models 
+with high performance.
+
+We recognize the challenges involved in operationalizing ML models performantly in an agile way, and we understand that high volume production services can be highly performance-sensitive and often need to support a variety of compute targets (we experience these first-hand at Microsoft across our vast array of products and services). 
+
+As such, our investments are directly in support of solving those challenges, focusing on areas such as:
```
Comment (pranavsharma) [inline docs/Roadmap.md]: Backwards compatibility is also one of the key challenges faced by teams when operationalizing models and one of the strongest plus point of ORT. It probably deserves a separate bullet of it's own where we mention not just backwards compatibility w.r.t models, but also of the APIs.

File: docs/Roadmap.md [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+We work with the OSS and ONNX community to ensure popular frameworks can export or be converted to ONNX format. 
+* [PyTorch export](https://pytorch.org/docs/stable/onnx.html)
+* [Tensorflow-ONNX](https://github.com/onnx/tensorflow-onnx)
+* [Keras-ONNX](https://github.com/onnx/keras-onnx)
+* [Sklearn-ONNX](https://github.com/onnx/sklearn-onnx)
+* [ONNXMLTools](https://github.com/onnx/onnxmltools/tree/master/onnxmltools/convert) (CoreML, XGBoost, LibSVM, LightGBM, SparkML)
+* [ML.NET](https://github.com/dotnet/machinelearning)
+
+#### Improved error handling
+To decrease the risk model inferencing failures, we will improve the error handling and fallback strategies for missing types or unsupported operators.
```
Comment (pranavsharma) [inline docs/Roadmap.md]: risk *of* model inferencing

File: docs/Roadmap.md [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+|C#| |
+|C| |
+|Ruby (community project)| |
+
+
+### Accelerators and Execution Providers
+#### New EPs
+To achieve the best performance on a growing set of compute targets across cloud and the intelligent edge, we invest in and partner with hardware partners and community members to add new execution providers. The flexible pluggability of ONNX Runtime is critical to support a broad range of scenarios and compute options.
+
+|Supported|Future|
```
Comment (pranavsharma) [inline docs/Roadmap.md]: Should we mention ACL (Arm Compute Library) EP since we've an active PR for it?

File: docs/Roadmap.md [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+#### Simplify EP contributions
+In addition to new execution providers, we aim to make it easy for community partners to contribute in a non-disruptive way. To support this, we are investing in improvements to the execution provider interface for easily registering new execution providers and separating out EPs from the core runtime engine.
+
+### Continued Performance Optimizations
+Performance is a key focus for ONNX Runtime. From latency to memory utilization to CPU usage, we are constantly seeking strategies to deliver the best performance. Although DNNs are rapidly driving research areas for innovation, we acknowledge that in practice, many companies and developers are still using traditional ML frameworks for reasons ranging from expertise to privacy to legality. As such, ONNX Runtime is focused on improvements and support for both DNNs and traditional ML. 
+
+#### Examples of projects the team is working on:
+* Improvements to batch processing for scikit-learn models
+* More quantization support
+* Multithreading
```
Comment (pranavsharma) [inline docs/Roadmap.md]: This is a bit too generic and might imply that we don't have multi-threading at all. Should we word it as - Improved multithreading including but not limited to smarter work sharding, consolidation of different types of thread pools, user supplied thread pools, etc.

File: docs/Roadmap.md [hunk 7]
```diff
@@ -0,0 +1,150 @@
+# ONNX Runtime Roadmap
+ONNX Runtime is an active, fast-paced project backed by a strong team of deeply technical Microsoft engineers along with a worldwide
```
Comment (pranavsharma) [inline docs/Roadmap.md]: suggest reword this as - strong team of engineers and data scientists

File: docs/Roadmap.md [hunk 8]
```diff
@@ -0,0 +1,150 @@
+# ONNX Runtime Roadmap
+ONNX Runtime is an active, fast-paced project backed by a strong team of deeply technical Microsoft engineers along with a worldwide 
+community of partners and contributors. This roadmap summarizes the pending investments identified by the team to continually grow 
+ONNX Runtime as a robust, versatile, and high performance engine for DNN and traditional ML models.
+
+## High Level Goals
+ONNX Runtime fully aligns with the goals of the [ONNX](https://onnx.ai/) community to bring openness and interoperability to the AI industry.
```
Comment (pranavsharma) [inline docs/Roadmap.md]: Also consider if we should add some of the high level goals from this [slide 35](https://microsoft.sharepoint.com/:p:/r/teams/ONNXDiscussions/_layouts/15/Doc.aspx?sourcedoc=%7B6348F0A6-BA59-464D-A52C-53C737E2B029%7D&file=Aug2019%20-%20ORT%20overview%20for%20partners.pptx&action=edit&mobileredirect=true).

File: docs/Roadmap.md [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+
+### Accelerators and Execution Providers
+#### New EPs
+To achieve the best performance on a growing set of compute targets across cloud and the intelligent edge, we invest in and partner with hardware partners and community members to add new execution providers. The flexible pluggability of ONNX Runtime is critical to support a broad range of scenarios and compute options.
+
+|Supported|Future|
+|---|---|
+|MLAS (Microsoft Linear Algebra Subprograms)|Android NN API (in progress)|
+|Intel MKL-DNN / MKL-ML|Qualcomm (TBD)|
+|Intel nGraph|AMD (TBD)|
```
Comment (manashgoswami) [inline docs/Roadmap.md]: Remove Qualcomm and AMD.
Add "ARM Compute Lib (by NXP)"

File: docs/Roadmap.md (hunk 4 above)
Comment (manashgoswami) [reply docs/Roadmap.md]: ... and ONNX spec compliance.

File: docs/Roadmap.md [hunk 10]
```diff
@@ -0,0 +1,150 @@
...
+### Continued Performance Optimizations
+Performance is a key focus for ONNX Runtime. From latency to memory utilization to CPU usage, we are constantly seeking strategies to deliver the best performance. Although DNNs are rapidly driving research areas for innovation, we acknowledge that in practice, many companies and developers are still using traditional ML frameworks for reasons ranging from expertise to privacy to legality. As such, ONNX Runtime is focused on improvements and support for both DNNs and traditional ML. 
+
+#### Examples of projects the team is working on:
+* Improvements to batch processing for scikit-learn models
+* More quantization support
+* Multithreading
+* Graph optimizations
+* Intelligent graph partitioning to maximize the value of different accelerators
+* Optimizations for mobile and IoT Edge devices
```
Comment (manashgoswami) [inline docs/Roadmap.md]: ### Optimizations for IoT:
IoT is a growing opportunity to execute ML workloads on the edge of the network, where data is collected. But the devices used for ML execution have different HW specifications. We will invest in areas for optimizing ONNX model execution across the breadth of IoT endpoints using different HW configurations with CPUs, GPUs and custom NN ASICs.

File: docs/Roadmap.md (hunk 8 above)
Comment (faxu) [reply docs/Roadmap.md]: I added a link to the technical design page, which covers some of those topics
